x-pack/plugins/infra/common/alerting/metrics/alert_link.ts

```diff
@@ -46,37 +49,66 @@ export const getInventoryViewInAppUrl = (
-      nodeType: inventoryFields[nodeTypeField][0],
-      timestamp: Date.parse(inventoryFields[TIMESTAMP]),
-      customMetric: '',
-    };
-    // We always pick the first criteria metric for the URL
-    const criteriaMetric = inventoryFields[`${ALERT_RULE_PARAMETERS}.criteria.metric`][0];
-    if (criteriaMetric === 'custom') {
-      const criteriaCustomMetricId =
-        inventoryFields[`${ALERT_RULE_PARAMETERS}.criteria.customMetric.id`][0];
-      const criteriaCustomMetricAggregation =
-        inventoryFields[`${ALERT_RULE_PARAMETERS}.criteria.customMetric.aggregation`][0];
-      const criteriaCustomMetricField =
-        inventoryFields[`${ALERT_RULE_PARAMETERS}.criteria.customMetric.field`][0];
-
-      const customMetric = encode({
-        id: criteriaCustomMetricId,
-        type: 'custom',
-        field: criteriaCustomMetricField,
-        aggregation: criteriaCustomMetricAggregation,
-      });
-      linkToParams.customMetric = customMetric;
-      linkToParams.metric = customMetric;
+    if (hostName) {
+      viewInAppUrl = getLinkToHostDetails(hostName, inventoryFields[TIMESTAMP]);
     } else {
-      linkToParams.metric = encode({ type: criteriaMetric });
+      const linkToParams: Record<string, any> = {
+        nodeType: inventoryFields[nodeTypeField][0],
+        timestamp: Date.parse(inventoryFields[TIMESTAMP]),
+        customMetric: '',
+      };
+
+      // We always pick the first criteria metric for the URL
+      const criteriaMetric = inventoryFields[`${ALERT_RULE_PARAMETERS}.criteria.metric`][0];
+      if (criteriaMetric === 'custom') {
+        const criteriaCustomMetricId =
+          inventoryFields[`${ALERT_RULE_PARAMETERS}.criteria.customMetric.id`][0];
+        const criteriaCustomMetricAggregation =
+          inventoryFields[`${ALERT_RULE_PARAMETERS}.criteria.customMetric.aggregation`][0];
+        const criteriaCustomMetricField =
+          inventoryFields[`${ALERT_RULE_PARAMETERS}.criteria.customMetric.field`][0];
+
+        const customMetric = encode({
+          id: criteriaCustomMetricId,
+          type: 'custom',
+          field: criteriaCustomMetricField,
+          aggregation: criteriaCustomMetricAggregation,
+        });
+        linkToParams.customMetric = customMetric;
+        linkToParams.metric = customMetric;
+      } else {
+        linkToParams.metric = encode({ type: criteriaMetric });
+      }
+      viewInAppUrl += stringify(linkToParams);
     }
-    inventoryViewInAppUrl += stringify(linkToParams);
   }
-  return inventoryViewInAppUrl;
+
+  return viewInAppUrl;
+};
+
+export const getMetricsViewInAppUrl = (fields: ParsedTechnicalFields & Record<string, any>) => {
+  const hostName = fields[HOST_FIELD];
+  const timestamp = fields[TIMESTAMP];
+
+  if (hostName) {
+    return getLinkToHostDetails(hostName, timestamp);
+  }
+  return LINK_TO_METRICS_EXPLORER;
 };
+
+export function getLinkToHostDetails(hostName: string, timestamp: string): string {
```

Comment:
To avoid mistakes when calling this function, what do you think of using obj params?
```suggestion
export function getLinkToHostDetails({hostName, timestamp}:{hostName: string, timestamp: string}) {
```

Reply:
what I was thinking when creating the function? 😆